### PR TITLE
edgemesh code decoupling

### PR DESCRIPTION
**Pull request by Poorunga**
According to the proposal: https://github.com/kubeedge/kubeedge/pull/2627, we have decoupled edgemesh from edgecore:
1. code transplantation comes from: https://github.com/Poorunga/edgemesh
2. update the readme.md file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,303 @@
...
+- Through the capability of list-watch on the edge of KubeEdge, EdgeMesh monitors the addition, deletion and modification of metadata (e.g., services and endpoints), and then creates iptables rules based on services and endpoints
+- EdgeMesh uses the same ways (e.g., Cluster IP, domain name) as the K8s Service to access services
+- When client's requests accessing a service reach a node with EdgeMesh, it will enter the kernel's iptables at first
+- The iptables rules previously configured by EdgeMesh will redirect requests, and forward them all to the port 40001 which is occupied by the EdgeMesh process (data packets from kernel mode to user mode)
+- After requests enter the EdgeMesh process, the EdgeMesh process completes the selection of backend Pods (load balancing occurs here), and then sends requests to the host where the Pod is located
+
+
+
+
+## Functionality
```
Comment (kevin-wangzefeng) [inline README.md:54]: Key features and roadmap

File: README.md [hunk 2]
```diff
@@ -1 +1,303 @@
...
+#### User value
+
+- For edge devices with limited resources, EdgeMesh provides a lightweight and highly integrated software with service discovery
+- In the scene of Field Edge, compared to the mechanism of coredns + kube-proxy + cni service discovery , users only need to simply deploy an EdgeMesh to finish their goals
+
+
+
+## Architecture
+
+![](/images/em-arch.png)
```
Comment (kevin-wangzefeng) [inline README.md:41]: Please redraw this diagram with latest project components

File: README.md [hunk 3]
```diff
@@ -1 +1,303 @@
...
\ No newline at end of file
+# EdgeMesh
+
+English | [简体中文](./README_ZH.md)
+
+
+
+## Introduction
+
+EdgeMesh is closely related to KubeEdge, and provides a simple network solution for the inter-communications between services at edge scenarios.
```
Comment (kevin-wangzefeng) [inline README.md:9]: EdgeMesh is part of KubeEdge

File: README.md [hunk 4]
```diff
@@ -1 +1,303 @@
...
+
+At present, EdgeMesh does not have a Pod network, so it relies on the connectivity of the host network. In the future, EdgeMesh will realize the capabilities of CNI plug-ins, and realize the Pod network connectivity between edge nodes and nodes on the cloud, or edge nodes across LANs in a  compatible manner with mainstream CNI plug-ins (e.g., flannel / calico, etc). Finally, EdgeMesh can even replace part of its own components with cloud-native components (e.g., replacing [kube-proxy](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/) to achieve the capabilities of the Cluster IP, replacing [node local dns cache ](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/) to achieve node-level dns capabilities, and replace [envoy](https://www.envoyproxy.io/) to achieve mesh-layer capabilities).
```
Comment (kevin-wangzefeng) [inline README.md:11]: Please describe the project background, motivation and target position (in the tech stack) here.

File: README.md [hunk 5]
```diff
@@ -1 +1,303 @@
...
+
+**Noting:**
+
+- `✓` Features supported by the EdgeMesh version 
+- `+` Features not available in the EdgeMesh version, but will be supported in subsequent versions
+- `-` Features not available in the EdgeMesh version, or deprecated features
+
+
+
+## Operation Guidance
```
Comment (kevin-wangzefeng) [inline README.md:77]: Getting started

File: README.md [hunk 6]
```diff
@@ -1 +1,303 @@
...
+
+#### Constraints
```
Comment (kevin-wangzefeng) [inline README.md:79]: Prerequisites

File: README.md [hunk 7]
```diff
@@ -1 +1,303 @@
...
+In its implementation, Edgemesh has borrowed from [istio's](https://istio.io/latest/zh/docs/ops/deployment/requirements/) VirtualService, DestinationRule, and GateWay, so there are some requirements in its use:
+
+- Due to the lack of underlying CNI capabilities, when using edgemesh's capabilities, the Pod is required a hostPort (as shown in following examples)
+- while using DestinationRule, the name of the DestinationRule must be equal to the name of the corresponding Service. Edgemesh will determine the DestinationRule in the same namespace according to the name of the Service
+- Service ports must be named. The key/value pairs of port name must have the following syntax: name: \<protocol>[-\<suffix>]
```
Comment (kevin-wangzefeng) [inline README.md:84]: just describe what need to be done before starting to use EdgeMesh

File: README.md (hunk 1 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

File: README.md (hunk 3 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

File: README.md (hunk 4 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

File: README.md (hunk 5 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

File: README.md (hunk 6 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

File: README.md (hunk 7 above)
Comment (Poorunga) [reply README.md]: I have modified this content as required.

**Review by fisherxu — COMMENTED** [review]
Seems have no makefile, so how to build the binary and image? Please add it later.

**Pull request merged**